ipn/ipnserver/server.go

```diff
@@ -59,7 +63,64 @@ func (s *Server) mustBackend() *ipnlocal.LocalBackend {
 	return lb
 }
 
+func (s *Server) awaitBackend(ctx context.Context) (_ *ipnlocal.LocalBackend, ok bool) {
+	lb := s.lb.Load()
+	if lb != nil {
+		return lb, true
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	s.mu.Lock()
+	h := s.backendWaiter.Add(cancel)
+	s.mu.Unlock()
+	defer func() {
+		s.mu.Lock()
+		delete(s.backendWaiter, h)
+		s.mu.Unlock()
+	}()
+
+	// Try again, now that we've registered, in case there was a
+	// race.
+	lb = s.lb.Load()
+	if lb != nil {
+		return lb, true
+	}
+
+	<-ctx.Done()
+	lb = s.lb.Load()
+	return lb, lb != nil
+}
+
+// serveServerStatus serves the /server-status endpoint which reports whether
+// the LocalBackend is up yet.
+// This is primarily for the Windows GUI, because wintun can take awhile to
+// come up. See https://github.com/tailscale/tailscale/issues/6522.
+func (s *Server) serveServerStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	w.Header().Set("Content-Type", "application/json")
+	var res struct {
+		Error string `json:"error,omitempty"`
+	}
+
+	lb := s.lb.Load()
+	if lb == nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		if wait, _ := strconv.ParseBool(r.FormValue("wait")); wait {
+			w.(http.Flusher).Flush()
+			lb, _ = s.awaitBackend(ctx)
+		}
```

Comment:
Do we want to write `http.StatusServiceUnavailable` even when waiting?

Reply:
Yeah, I figured that'd be the signal that "we're actually waiting".

Admittedly this isn't a very defined protocol.